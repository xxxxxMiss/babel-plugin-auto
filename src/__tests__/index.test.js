import * as babel from '@babel/core'
import plugin from '../'

const str = `
import styles from 'index.less'

const x = {
  o: cx('name')
}
cx('hello-world')
const y = cx('test')
`

test('should works', () => {
  const { code } = babel.transform(str, {
    plugins: [plugin],
    parserOpts: {
      plugins: ['jsx'],
    },
  })
  expect(code).toMatchSnapshot()
})
