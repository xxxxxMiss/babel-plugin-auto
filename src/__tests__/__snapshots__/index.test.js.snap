// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should works 1`] = `
"\\"use strict\\";

var _index = _interopRequireDefault(require(\\"index.less\\"));

var _bind = _interopRequireDefault(require(\\"classnames/bind\\"));

var cx = _bind.default.bind(_index.default);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_index.default);

const x = {
  o: _bind.default.bind(_index.default)('name')
};

_bind.default.bind(_index.default)('hello-world');

const y = _bind.default.bind(_index.default)('test');"
`;
